pkg/cbox/share/sql/sql.go

```diff
@@ -464,3 +486,69 @@ func (m *mgr) UpdateReceivedShare(ctx context.Context, ref *collaboration.ShareR
 	rs.State = f.GetState()
 	return rs, nil
 }
+
+func groupFiltersByType(filters []*collaboration.Filter) map[collaboration.Filter_Type][]*collaboration.Filter {
+	sorted := make(map[collaboration.Filter_Type][]*collaboration.Filter)
+	for _, f := range filters {
+		sorted[f.Type] = append(sorted[f.Type], f)
+	}
+	return sorted
+}
+
+func granteeTypeToShareType(granteeType provider.GranteeType) int {
+	switch granteeType {
+	case provider.GranteeType_GRANTEE_TYPE_USER:
+		return shareTypeUser
+	case provider.GranteeType_GRANTEE_TYPE_GROUP:
+		return shareTypeGroup
+	}
+	return -1
```

Comment:
why not using multiple returns and actually return an error here?

Reply:
or is this `-1` [here](https://github.com/cs3org/reva/pull/2050/files#diff-bee286ddbc20ada3dd35cb24990f6044dc9809ef6243519e28697ea3ca7be184R538) accepted?